Create a combined GTK/WPE PlatformEnablement file

brentfulgham opened the pull request:
#### 73e0d4c74d1baaeb169b9bbc3f1adcfe71579bcc
<pre>
Create a combined GTK/WPE PlatformEnablement file
<a href="https://bugs.webkit.org/show_bug.cgi?id=261960">https://bugs.webkit.org/show_bug.cgi?id=261960</a>
&lt;rdar://problem/115902210&gt;

Reviewed by Chris Dumez.

PlatformEnable.h is meant to hold the default state of compile-time feature enablement
flag. Specific platform-related files are meant to override these values.

No PlatformEnableGTK or PlatformEnableWPE file was ever created, leading to conditional
macros being added to the default file.

This patch moves the GTK and WPE-specific rules to their own file, to match other platforms.
It also corrects the alphabetical ordering of the files.

* Source/WTF/wtf/CMakeFiles.txt: Add new header file.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformEnableGlib.h: Added.
* Source/WTF/wtf/PlatformEnablePlayStation.h:
* Source/WTF/wtf/PlatformEnableWinCairo.h:
* Source/WebCore/platform/PlatformWheelEvent.h: Correct the enum definition for the
case where KINETIC_SCROLLING has been enabled, but ASYNC_SCROLLING is not.

Canonical link: <a href="https://commits.webkit.org/268543@main">https://commits.webkit.org/268543@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c9aface2acd782a8038c286a9abddd991d62ad01

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19981 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20405 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21028 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21876 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18664 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20214 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23660 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20558 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20170 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20201 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20154 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17380 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22728 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17324 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18176 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24447 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/17382 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18401 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18352 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22436 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/19352 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18937 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16065 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/23384 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18118 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Tests-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4782 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22467 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/24640 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18764 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5450 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WTF/WTF.xcodeproj/project.pbxproj:
```diff
@@ -109,6 +109,7 @@
 		70A993FE1AD7151300FA615B /* SymbolRegistry.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 70A993FC1AD7151300FA615B /* SymbolRegistry.cpp */; };
 		70ECA60D1B02426800449739 /* AtomStringImpl.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 70ECA60A1B02426800449739 /* AtomStringImpl.cpp */; };
 		7A05093F1FB9DCC500B33FB8 /* JSONValues.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 7A05093E1FB9DCC500B33FB8 /* JSONValues.cpp */; };
+		7A4228942ABE0FF60065F7B5 /* PlatformEnableGTKWPE.h in Headers */ = {isa = PBXBuildFile; fileRef = 7A4228932ABE0FF60065F7B5 /* PlatformEnableGTKWPE.h */; };
```
It'd be nice to use "Glib" instead of "GTKWPE". "glib" is already the abbreviation used in layout tests to mean both GTK and WPE.

brentfulgham replied:
That's a great idea -- I'll make that change and run through EWS.

brentfulgham committed:
Create a combined GTK/WPE PlatformEnablement file
https://bugs.webkit.org/show_bug.cgi?id=261960
<rdar://problem/115902210>

Reviewed by Chris Dumez.

PlatformEnable.h is meant to hold the default state of compile-time feature enablement
flag. Specific platform-related files are meant to override these values.

No PlatformEnableGTK or PlatformEnableWPE file was ever created, leading to conditional
macros being added to the default file.

This patch moves the GTK and WPE-specific rules to their own file, to match other platforms.
It also corrects the alphabetical ordering of the files.

* Source/WTF/wtf/CMakeFiles.txt: Add new header file.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformEnableGlib.h: Added.
* Source/WTF/wtf/PlatformEnablePlayStation.h:
* Source/WTF/wtf/PlatformEnableWinCairo.h:
* Source/WebCore/platform/PlatformWheelEvent.h: Correct the enum definition for the
case where KINETIC_SCROLLING has been enabled, but ASYNC_SCROLLING is not.

Canonical link: https://commits.webkit.org/268543@main
